Moonscraper Chart Editor/Assets/Scripts/Game/UI/Menus/SongPropertiesPanelController.cs

```diff
@@ -320,6 +327,58 @@ public void RefreshAllAudioStreams()
         StartCoroutine(_RefreshAllAudioStreams());
     }
 
+    public void GetInstrumentAudioByFileNames()
+    {
+        // TODO: there HAS to be a way to just detect the current folder, right???
+        FileExplorer.OpenFolderPanel(out string resultPath);
+        var filesInDir = Directory.GetFiles(resultPath, "*.*");
+        foreach (var filePath in filesInDir.Where(x => validAudioExtensions.Contains(Path.GetExtension(x).TrimStart('.'))))
+        {
+            Song.AudioInstrument instrument;
+            switch(Path.GetFileNameWithoutExtension(filePath)) {
+                case "song":
+                    instrument = Song.AudioInstrument.Song;
+                    break;
+                case "guitar":
+                    instrument = Song.AudioInstrument.Guitar;
+                    break;
+                case "bass":
+                    instrument = Song.AudioInstrument.Bass;
+                    break;
+                case "rhythm":
+                    instrument = Song.AudioInstrument.Rhythm;
+                    break;
+                case "vocals":
+                    instrument = Song.AudioInstrument.Vocals;
+                    break;
+                case "drums":
+                case "drums_1":
+                    instrument = Song.AudioInstrument.Drum;
+                    break;
+                case "drums_2":
+                    instrument = Song.AudioInstrument.Drums_2;
+                    break;
+                case "drums_3":
+                    instrument = Song.AudioInstrument.Drums_3;
+                    break;
+                case "drums_4":
+                    instrument = Song.AudioInstrument.Drums_4;
+                    break;
+                case "keys":
+                    instrument = Song.AudioInstrument.Keys;
+                    break;
+                case "crowd":
+                    instrument = Song.AudioInstrument.Crowd;
+                    break;
+                default:
+                    // would like to use a switch expression, but since this enum is non-nullable,
+                    // gotta use a continue to skip unwanted files
+                    continue;
```

Comment:
This would be better to have a list of the files that were skipped over, and then communicated to the user of which files were invalid and what the correct filenames should be (can send a message via ChartEditor.Instance.errorManager.QueueErrorMessage)

Reply:
@FireFox2000000 are there any other filenames that should be considered non-error-worthy, but not be tied to an instrument? e.g. i know "preview.ogg" is used for separate song preview audio files in Clone Hero

Reply:
Not that I'm aware of.